File: src/main/java/cc/suvankar/moneytrail/contact/dto/ContactRequest.java
package cc.suvankar.moneytrail.contact.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ContactRequest {
  @NotBlank(message = "Name is required.")
  private String name;

  @Email(regexp = ".+@.+\\..+", message = "Must be a valid email address")
  private String email;

  @Pattern(regexp = "^\\+?[1-9]\\d{1,14}$", message = "Invalid phone number format")
  private String phoneNo;

  private String notes;
}

File: src/main/kotlin/cc/suvankar/moneytrail/contact/dto/ContactRequest.kt
package cc.suvankar.moneytrail.contact.dto

import jakarta.validation.constraints.*

data class ContactRequest(
  @field:NotBlank("Name is required.") val name: String,
  @field:Email(regexp = ".+@.+\\..+", message = "Must be a valid email address") val email: String?,
  @field:Pattern(regexp = "^\\+?[1-9]\\d{1,14}$", message = "Invalid phone number format")
  val phoneNo: String?,
  val notes: String?,
)
